Source/Core/InputCommon/ControlReference/ControlReference.h

```diff
@@ -52,11 +52,17 @@ class ControlReference
 template <>
 inline bool ControlReference::GetState<bool>()
 {
-  return State() > ciface::ExpressionParser::CONDITION_THRESHOLD;
+  return std::lround(State()) > 0;
```

Comment:
I'd be careful with std::lround(), GCC is the only compiler that optimizes this to a movsx. Clang and MSVC both emit a float conversion and a call into the stdlib.

Reply:
But now I'm wondering why you convert to long.

Reply:
@Tilka I'm just using `std::lround` to round to the nearest of 0 or 1.
That seemed cleaner to me than a 0.5 constant.

Reply:
Checking the value directly seems clearer to me, but I'm not the one who maintains this code^^

Reply:
I'll add a comment so it's clear what's happening.